File: go/config/Device.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Device 对应 config/Device.py。
type Device struct {
	*session.ApiGroup
}

// NewDevice 创建 Device API 分组。
func NewDevice(s *session.Session) *Device {
	return &Device{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 device/config.xml。
func (d *Device) Config() (map[string]interface{}, error) {
	return d.S.Get("device/config.xml", nil, "config")
}

File: src/config/device.rs
//! Device config group (`config/Device.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Device config group.
pub struct DeviceConfig<'a> {
    conn: &'a Connection,
}

impl<'a> DeviceConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        DeviceConfig { conn }
    }

    /// `device/config.xml`. Device configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/device/config.xml")
    }
}
